### PR TITLE
Add bitswap metrics

### LINKED ISSUE #2679: Add Bitswap Metrics
Add metrics for Bitswap based on the following methods:

* [PeerConnected](https://pkg.go.dev/github.com/ipfs/go-bitswap#Bitswap.PeerConnected) / [PeerDisconnected](https://pkg.go.dev/github.com/ipfs/go-bitswap#Bitswap.PeerDisconnected): Track the total number of peers on Bitswap.
* [Stat](https://pkg.go.dev/github.com/ipfs/go-bitswap#Bitswap.Stat): Tracks various stats on Bitswap operations.
* [GetWantBlocks](https://pkg.go.dev/github.com/ipfs/go-bitswap#Bitswap.GetWantBlocks) / [GetWantHaves](https://pkg.go.dev/github.com/ipfs/go-bitswap#Bitswap.GetWantHaves): Tracks our total number of outstanding requests.
* [Tracer](https://pkg.go.dev/github.com/ipfs/go-bitswap#Tracer): Can be used to implement various statistics.

### PR DESCRIPTION
**Pull request by synzhu**
closes https://github.com/onflow/flow-go/issues/2679

### CODE REVIEW

**Review by kc1116 — APPROVED** [review]

File: network/p2p/blob_service.go [hunk 1]
```diff
@@ -84,9 +86,34 @@ func NewBlobService(
...
+
+			ticker := time.NewTicker(15 * time.Second)
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-ticker.C:
+					stat, err := btswp.Stat()
+					if err != nil {
+						// TODO: log error
```
Comment (peterargue) [inline network/p2p/blob_service.go:102]: please add this

**Review by peterargue — APPROVED** [review]
looks good. added one comment about logging

File: network/p2p/blob_service.go [hunk 2]
```diff
@@ -84,9 +86,34 @@ func NewBlobService(
...
+						continue
+					}
+
+					metrics.Peers(prefix, len(stat.Peers))
```
Comment (SaveTheRbtz) [inline network/p2p/blob_service.go:106]: it looks like prefix should be a property of the BitswapCollector instead of being passed to every function call.

**Comment by smnzhu** [comment]
bors merge

**Comment by smnzhu** [comment]
bors merge

File: cmd/util/ledger/reporters/mock/report_writer.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.13.1. DO NOT EDIT.
+// Code generated by mockery v2.13.0. DO NOT EDIT.
```
Comment (zhangchiqing) [inline cmd/util/ledger/reporters/mock/report_writer.go:1]: I think you need to upgrade your mockery to `2.13.1`

**Review by zhangchiqing — CHANGES_REQUESTED** [review]
please upgrade your mockery to 2.13.1

see instruction here https://github.com/onflow/flow-go/blob/master/Makefile#L61

**Review by zhangchiqing — APPROVED** [review]

**Comment by smnzhu** [comment]
bors merge

**Comment by smnzhu** [comment]
bors merge

**Pull request merged**